libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -331,7 +337,16 @@ method rpcHandler*(g: GossipSub,
     # g.anonymize needs no evaluation when receiving messages
     # as we have a "lax" policy and allow signed messages
 
+    # Be careful not to fill the validationSeen table
+    # (eg, pop everything you put in it)
+    g.validationSeen[msgIdSalted] = toHashSet([peer])
```

Comment:
wouldn't this override the previous entries?

Reply:
This code is only triggered for the first instance we receive of a message. As soon as we called `addSeen` above, every peer sending us this message will fail that condition (since we already saw it)

Reply:
How does this makes sense then, if we only do this once per message, then there is only ever going to be one peer per msg id? Seem a bit convoluted just to count one peer? What am I missing?

Reply:
Full code:
```nim
if g.addSeen(msgId):
  trace "Dropping already-seen message", msgId = shortLog(msgId), peer

  if msgIdSalted in g.validationSeen:
    g.validationSeen[msgIdSalted].incl(peer)
  
[...]
g.validationSeen[msgIdSalted] = toHashSet([peer])
```